pretix_newsletter_manual/exporters.py

```diff
@@ -29,12 +26,17 @@ def iterate_list(self, form_data):
                     headers.append(label)
         yield headers
 
-
         for r in qs.select_related("order", "order__event"):
             tz = pytz.timezone(r.order.event.settings.timezone)
-            possible_positions = [op for op in r.order.all_positions.all() if not op.canceled and op.attendee_name]
-            name_parts = possible_positions[0] if possible_positions else {}
-            
+            op_with_attendee_name = next(
+                (
+                    op
+                    for op in r.order.all_positions.all()
```

Comment:
@raphaelm Looking up the source code, it says that `all_positions` should be used, if you want canceled positions as well, but we filter them lateron in python. Is this because using `positions` would not be possible in `.prefetch_related` and thus using `positions` here would cause a query for every iteration of the loop?

Reply:
Correct!